common/middleware/src/meta_rules.rs

```diff
@@ -0,0 +1,516 @@
+            host: vars.get(LOGDNA_META_HOST).cloned(),
+            env: vars.get(LOGDNA_META_ENV).cloned(),
+            file: vars.get(LOGDNA_META_FILE).cloned(),
+            k8s_file: vars.get(LOGDNA_META_K8S_FILE).cloned(),
+            meta: vars.get(LOGDNA_META_JSON).cloned(),
+            annotations: vars.get(LOGDNA_META_ANNOTATIONS).cloned(),
+            labels: vars.get(LOGDNA_META_LABELS).cloned(),
+        }
+    }
+}
+
+pub struct MetaRules {
+    env_map: HashMap<String, String>,
+    // "override" fields
+    over_app: Option<String>,
+    over_host: Option<String>,
+    over_env: Option<String>,
+    over_file: Option<String>,
+    over_k8s_file: Option<String>, // k8s lines only
+    over_meta: Option<String>,
+    over_annotations: Option<HashMap<String, String>>, // "merge override", for  "delete and then override" - disable k8s enrichment
+    over_labels: Option<HashMap<String, String>>,      // --//--
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum MetaRulesError {
+    #[error(transparent)]
+    Serde(#[from] serde_json::Error),
+}
+
+impl MetaRules {
+    pub fn new(cfg: MetaRulesConfig) -> Result<MetaRules, MetaRulesError> {
+        let obj = MetaRules {
+            env_map: os_env_hashmap(),
+            over_app: cfg.app,
+            over_host: cfg.host,
+            over_env: cfg.env,
+            over_file: cfg.file,
+            over_k8s_file: cfg.k8s_file,
+            over_meta: cfg.meta,
+            over_annotations: cfg
+                .annotations
+                .map_or_else(|| None, |str| serde_json::from_str(str.as_str()).unwrap()),
+            over_labels: cfg
+                .labels
+                .map_or_else(|| None, |str| serde_json::from_str(str.as_str()).unwrap()),
+        };
+        Ok(obj)
+    }
+
+    pub fn is_active(&self) -> bool {
+        self.over_app.is_some()
+            || self.over_host.is_some()
+            || self.over_env.is_some()
+            || self.over_file.is_some()
+            || self.over_k8s_file.is_some()
+            || self.over_meta.is_some()
+            || self.over_annotations.is_some()
+            || self.over_labels.is_some()
+    }
+
+    /// process_line
+    /// - override line meta fields
+    /// [ create map ] => [ substitute then insert to map ] => [ substitute from map ] => [ override ]
+    ///  os env vars       "override" labels                    "override" fields          line fields
+    ///  line fields       "override" annotations
+    ///  line labels
+    ///  line annotations
+    /// Excluded:
+    ///  line "meta"
+    ///
+    fn process_line<'a>(
+        &self,
+        line: &'a mut dyn LineBufferMut,
+    ) -> Status<&'a mut dyn LineBufferMut> {
+        if !self.is_active() {
+            return Status::Ok(line);
+        }
+        //
+        // create map
+        //
+        let mut meta_map: HashMap<String, String> = self.env_map.clone();
+        if line.get_annotations().is_some() {
```

Comment:
Anywhere we're using `is_some` followed by an unwrap we could refactor to use if let, eg

```rust
        if let Some(annotations) = line.get_annotations().as_ref() {
            for (k, v) in annotations.iter() {
                meta_map.insert(k.clone(), v.clone());
            }
        }
```

Reply:
+1. updating.